Dockerfile

```
@@ -22,147 +22,122 @@
-    --exclude "fsl/etc/fslconf" \
-    --exclude "fsl/etc/js" \
-    --exclude "fsl/etc/luts" \
-    --exclude "fsl/etc/matlab" \
-    --exclude "fsl/extras" \
-    --exclude "fsl/include" \
-    --exclude "fsl/python" \
-    --exclude "fsl/refdoc" \
-    --exclude "fsl/src" \
-    --exclude "fsl/tcl" \
-    --exclude "fsl/bin/FSLeyes" \
-    && find /opt/fsl-6.0.5.1/bin -type f -not \( \
-        -name "applywarp" -or \
-        -name "bet" -or \
-        -name "bet2" -or \
-        -name "convert_xfm" -or \
-        -name "fast" -or \
-        -name "flirt" -or \
-        -name "fsl_regfilt" -or \
-        -name "fslhd" -or \
-        -name "fslinfo" -or \
-        -name "fslmaths" -or \
-        -name "fslmerge" -or \
-        -name "fslroi" -or \
-        -name "fslsplit" -or \
-        -name "fslstats" -or \
-        -name "imtest" -or \
-        -name "mcflirt" -or \
-        -name "melodic" -or \
-        -name "prelude" -or \
-        -name "remove_ext" -or \
-        -name "susan" -or \
-        -name "topup" -or \
-        -name "zeropad" \) -delete \
-    && find /opt/fsl-6.0.5.1/data/standard -type f -not -name "MNI152_T1_2mm_brain.nii.gz" -delete
-ENV FSLDIR="/opt/fsl-6.0.5.1" \
-    PATH="/opt/fsl-6.0.5.1/bin:$PATH" \
-    FSLOUTPUTTYPE="NIFTI_GZ" \
-    FSLMULTIFILEQUIT="TRUE" \
-    FSLLOCKDIR="" \
-    FSLMACHINELIST="" \
-    FSLREMOTECALL="" \
-    FSLGECUDAQ="cuda.q" \
-    LD_LIBRARY_PATH="/opt/fsl-6.0.5.1/lib:$LD_LIBRARY_PATH"
+# ANTs 2.3.3
+FROM downloader as ants
+# Note: the URL says 2.3.4 but it is actually 2.3.3
+RUN mkdir /opt/ants && \
+    curl -sSL "https://dl.dropbox.com/s/gwf51ykkk5bifyj/ants-Linux-centos6_x86_64-v2.3.4.tar.gz" \
+    | tar -xzC /opt/ants --strip-components 1
+
+# Connectome Workbench 1.5.0
+FROM downloader as workbench
+RUN mkdir /opt/workbench && \
+    curl -sSLO https://www.humanconnectome.org/storage/app/media/workbench/workbench-linux64-v1.5.0.zip && \
+    unzip workbench-linux64-v1.5.0.zip -d /opt && \
+    rm workbench-linux64-v1.5.0.zip && \
+    rm -rf /opt/workbench/libs_linux64_software_opengl /opt/workbench/plugins_linux64 && \
+    strip --remove-section=.note.ABI-tag /opt/workbench/libs_linux64/libQt5Core.so.5
+
+# Micromamba
+FROM downloader as micromamba
+WORKDIR /
+# Bump the date to current to force update micromamba
+RUN echo "2023.04.05"
+RUN curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba
+
+ENV MAMBA_ROOT_PREFIX="/opt/conda"
+COPY env.yml /tmp/env.yml
+RUN micromamba create -y -f /tmp/env.yml && \
+    micromamba clean -y -a
+
+ENV PATH="/opt/conda/envs/fmriprep/bin:$PATH"
+RUN /opt/conda/envs/fmriprep/bin/npm install -g svgo@^2.8 bids-validator@1.11.0 && \
+    rm -r ~/.npm
+
+COPY requirements.txt /tmp/requirements.txt
+RUN /opt/conda/envs/fmriprep/bin/pip install --no-cache-dir -r /tmp/requirements.txt
+
+#
+# Main stage
+#
+
+# Use Ubuntu 22.04 LTS
+FROM ubuntu:jammy-20221130 as fmriprep
```

then we could avoid any desync here
```suggestion
FROM ${BASE_OS} as fmriprep
```